lib/unexpectedMitm.js

```diff
@@ -730,6 +730,46 @@ module.exports = {
                     return recordedExchanges;
                 });
             })
+            .addAssertion('<any> with http mocked out [and verified] [returning the file] <string> <assertion>', function (expect, subject, filePathWithoutExt) {
+                expect.errorMode = 'nested';
+                var testFile = filePathWithoutExt + '.js';
+
+                if (process.env.UNEXPECTED_MITM_WRITE === 'true') {
+                    return executeMitm(expect, subject).then(function (recordedExchanges) {
+                        var output = 'module.exports = ' +  stringify(recordedExchanges) + ';\n';
+
+                        fs.writeFileSync(testFile, output);
+
+                        if (expect.flags['returning the file']) {
+                            return [recordedExchanges, testFile];
+                        } else {
+                            return recordedExchanges;
+                        }
+                    });
+                }
+
+                return expect.promise(function () {
+                    return require(testFile);
```

Comment:
This would be slightly better:

```js
var requireable = /^[./]/.test(testFile) ? testFile : './' + testFile;
return module.parent.require(requireable);
```

... but in the "unexpected-with-plugins.js" case `module.parent` will point at that, and resolving the relative paths from that won't necessarily be equivalent to resolving them from the actual test file.

Maybe the the entire chain of parent modules as "search paths" would work better (untested):

```js
for (var m = module.parent ; m ; m = m.parent) {
    try {
        return m.require(requirable);
    } catch (err) {
        if (err.code !== 'MODULE_NOT_FOUND' || !m.parent) {
            throw err;
        }
    }
}
```